ethereum/sidecar/server.go

```diff
@@ -53,12 +67,35 @@ var (
 	errInvalidEvents = fmt.Errorf("invalid AssetsLocked events")
 )
 
-// Server observes events emitted by the Mezo `MezoBridge` contract on the
-// Ethereum chain. It enables retrieval of information on the assets locked by
-// the contract. It is intended to be run as a separate process.
+// BridgeOutClient is a client enabling communication with the `Mezo` chain.
+type BridgeOutClient interface {
+	// GetAssetsUnlockedSequenceTip gets the assets unlocked sequence tip from
+	// the Mezo chain. The returned sequence tip is equal to the number of
+	// AssetsUnlocked events made so far. It is also equal to the value of the
+	// unlock sequence in the newest AssetsUnlocked event.
+	GetAssetsUnlockedSequenceTip(ctx context.Context) (sdkmath.Int, error)
+
+	// GetAssetsUnlockedEvents gets the AssetsUnlocked events from the Mezo
+	// chain. The requested range of events is inclusive on the lower side and
+	// exclusive on the upper side.
+	GetAssetsUnlockedEvents(
+		ctx context.Context,
+		sequenceStart sdkmath.Int,
+		sequenceEnd sdkmath.Int,
+	) ([]bridgetypes.AssetsUnlockedEvent, error)
+}
+
+// TimeFunc is a function that returns the current time.
+type TimeFunc func() time.Time
```

Comment:
I'm assuming this is done for testing purpose. I've usually been used to instead create a time package (e.g: mezotime) in the project I work on, where I'd define something like:
```
var Now = time.Now
```

Then in the code use mezotime.Now() when need the time. during normal execution of the software, nothing to do, but when testing, before your test, you can just easily replace the function with a mocked one. Not saying it's a better solution or anything though.

Reply:
Done in 12108804b86b4f85a3fbf8e33f80eaf3e4188ac4.
I added the `mezotime` package as you suggested.